lib/util.js

```diff
@@ -738,7 +738,9 @@ function formatSet(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
   value.forEach(function(v) {
     var nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-    var str = formatValue(ctx, v, nextRecurseTimes);
+    var str = ctx.seen.includes(v) ?
```

Comment:
This isn't directly related to this PR, but why is `ctx.seen` an array? Would it be better to use a `Set`/`WeakSet` to avoid linear-time `includes` checks?